modules/rpc/yarpc.go

```diff
@@ -249,7 +269,24 @@ func newYARPCModule(
 	module.config.inboundMiddleware = inboundMiddlewareFromCreateInfo(mi)
 	module.config.onewayInboundMiddleware = onewayInboundMiddlewareFromCreateInfo(mi)
 
-	_controller.addConfig(module.config)
+	// Try to resolve a controller first
+	// TODO(alsam) use dig options when available, because we can overwrite the controller in case of multiple
+	// modules registering a controller.
+	if err := dig.Resolve(&module.controller); err != nil {
```

Comment:
There can be multiple modules that can can use YARPC, dispatcherController combines their configs in one and creates a single dispatcher, that a service will use.